### PR TITLE
Feature/033 add markdown redactor

### PR DESCRIPTION
**Pull request by PavelOlkhovoi**

### CODE REVIEW

File: src/components/mdredactor/MdRedactor.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  btnStrikethrough: "Durchgestrichen",
+  btnUnordered: "Aufzählungsliste",
+  btnOrdered: "Nummerierte Liste",
+  btnQuote: "Zitat",
+  btnLink: "Link",
+});
+MdEditor.useLocale("de-DE");
+
+const ViewMode = (props) => {
+  const [isWriteActive, setIsWriteActive] = useState(true);
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: You have 2 States that basically do the same thing. Remove isWriteActive state

File: src/components/mdredactor/MdRedactor.jsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      html: false,
+    });
+    setIsWriteActive(true);
+    setIsPreviewActive(false);
+  };
+
+  return (
+    <div>
+      <div
+        className={isWriteActive ? "mode-btn-toogle-active" : "mode-btn-toogle"}
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: !isPreviewActive instead of isWriteActive

File: src/components/mdredactor/MdRedactor.jsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        onClick={handleClickWrite}
+        // title="Bearbeitenmodus"
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: remove comment

File: src/components/mdredactor/MdRedactor.jsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+      >
+        Bearbeiten
+      </div>
+      <div
+        className={
+          isPreviewActive ? "mode-btn-toogle-active" : "mode-btn-toogle"
+        }
+        onClick={handleClickPreview}
+        // title="Vorschaumodus"
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: remove comment

File: src/components/mdredactor/MdRedactor.jsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+ViewMode.defaultConfig = {};
+ViewMode.align = "right";
+ViewMode.pluginName = "viewmode";
+
+MdEditor.use(ViewMode, {});
+
+const SaveDocumentBtn = (props) => {
+  const handleSaveDocument = () => {
+    const mdDoc = props.editor.getMdValue();
+    console.log("xxx mdDoc", { mdDoc });
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: remove console log

File: src/components/mdredactor/MdRedactor.jsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      // className="button"
+      >
+        <DeliveredProcedureOutlined
+          className="save-button"
+          title="Dokument speichern"
+          onClick={handleSaveDocument}
+          style={{ marginTop: "6px", marginLeft: "5px", cursor: "pointer" }}
+        />
+      </span>
+    </div>
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: we probably don't need that button but if we would don't put the onClick on the icon and remove the span. either use <button></button> with the onClick or use <div role="button" onClick={}></div> and move the classname to the button itself and not the icon. You also use style + className for the icon. Stick to one or the other but not both if you don't need to

File: src/components/mdredactor/MdRedactor.jsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+  "block-quote",
+  "link",
+  // "divider",
+  // "savedoc",
+  //   "image",
+];
+
+const MdRedactor = ({
+  mdDoc = "",
+  getDocFn = () => console.log("getDocFn function"),
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: We already know that this is a function when we look at the default value. No need to put Fn at the end of the prop name. I would just name it getDocument or whatever this is doing

File: src/components/mdredactor/MdRedactor.jsx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+  saveTrigger = 1,
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: What does it do? Use a better name so somebody who wants to use your comment knows what this does

File: src/components/mdredactor/MdRedactor.jsx [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+  height = "700px",
+}) => {
+  const [mdText, setMdText] = useState(mdDoc);
+
+  SaveDocumentBtn.defaultConfig = {
+    getDocFn,
+  };
+
+  const handleEditorChange = ({ html, text }) => {
+    console.log("xxx mdParser", text);
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: remove log

File: src/components/mdredactor/MdRedactor.jsx (hunk 8 above)
Comment (PavelOlkhovoi) [reply src/components/mdredactor/MdRedactor.jsx]: I want to use some trigger that, when changed, will initiate a function to save the document. Any ideas on the best way to implement this?

File: src/components/mdredactor/MdRedactor.jsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+    props.editor.setView({
+      md: true,
+      menu: true,
+      html: false,
+    });
+    setIsPreviewActive(false);
+  };
+
+  return (
+    <div>
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: you don't need the div here. can use <></> instead

File: src/components/mdredactor/MdRedactor.jsx [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+  "list-unordered",
+  "list-ordered",
+  "block-quote",
+  "link",
+];
+
+const MdRedactor = ({
+  mdDoc = "",
+  getDocument = () => console.log("getDoc function"),
+  saveTrigger = 1,
```
Comment (d4v3000) [inline src/components/mdredactor/MdRedactor.jsx]: saveTrigger is now unused and can be removed

**Pull request merged**